packages/fiori/src/MediaGallery.js

```diff
@@ -0,0 +1,568 @@
+import Integer from "@ui5/webcomponents-base/dist/types/Integer.js";
+import NavigationMode from "@ui5/webcomponents-base/dist/types/NavigationMode.js";
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import Button from "@ui5/webcomponents/dist/Button.js";
+import Carousel from "@ui5/webcomponents/dist/Carousel.js";
+// Template
+import MediaGalleryTemplate from "./generated/templates/MediaGalleryTemplate.lit.js";
+// Styles
+import MediaGalleryCss from "./generated/themes/MediaGallery.css.js";
+import MediaGalleryItem from "./MediaGalleryItem.js";
+import MediaGalleryItemLayoutType from "./types/MediaGalleryItemLayoutType.js";
+import MediaGalleryLayoutType from "./types/MediaGalleryLayoutType.js";
+import MediaGalleryMenuHorizontalAlign from "./types/MediaGalleryMenuHorizontalAlign.js";
+import MediaGalleryMenuVerticalAlign from "./types/MediaGalleryMenuVerticalAlign.js";
+
+// The allowed number of thumbnail columns on each size
+// (relevant when <code>showAllThumbnails</code> is enabled)
+const COLUMNS_COUNT = {
+	"S": 1,
+	"M": 2,
+	"L": 3,
+	"XL": 4,
+};
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-media-gallery",
+	managedSlots: true,
+	slots: /** @lends  sap.ui.webcomponents.fiori.MediaGallery.prototype */ {
+		/**
+		 * Defines the component items.
+		 *
+		 * <br><br>
+		 * <b>Note:</b> Only one selected item is allowed.
+		 *
+		 * <br><br>
+		 * <b>Note:</b> Use the <code>ui5-media-gallery-item</code> component to define the desired items.
+		 * @type {sap.ui.webcomponents.fiori.IMediaGalleryItem[]}
+		 * @slot items
+		 * @public
+		 */
+		 "default": {
+			propertyName: "items",
+			type: HTMLElement,
+			individualSlots: true,
+			invalidateOnChildChange: true,
+		},
+	},
+	properties: /** @lends  sap.ui.webcomponents.fiori.MediaGallery.prototype */ {
+		/**
+		 * If set to <code>true</code>, all thumbnails will be rendered in a scrollable container.
+		 * If <code>false</code>, only up to 5 thumbnails will be rendered, followed by
+		 * an overflow button that shows the count of the remaining thumbnails.
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		 showAllThumbnails: {
+			type: Boolean,
+		},
+
+		/**
+		 * If enabled, a <code>display-area-click</code> event will be fired
+		 * when the user clicks on the display area.
+		 * <br>
+		 * The display area is the central area that contains
+		 * the enlarged content of the currently selected item.
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		 interactiveDisplayArea: {
+			type: Boolean,
+		},
+
+		/**
+		 * Determines the layout of the component.
+		 * <br><br>
+		 * Available options are:
+		 * <ul>
+		 * <li><code>Auto</code></li>
+		 * <li><code>Vertical</code></li>
+		 * <li><code>Horizontal</code></li>
+		 * </ul>
+		 * @type {MediaGalleryLayoutType}
+		 * @defaultvalue "Auto"
+		 * @public
+		 */
+		 layoutType: {
```

Comment:
Laving as topic to discuss: **layoutType** or  just **layout**

Reply:
Refactored to just "layout"